Ghidra/Features/GnuDemangler/src/main/java/ghidra/app/util/demangler/gnu/GnuDemanglerOptions.java

```diff
@@ -43,61 +43,133 @@
+		this.deprecated = format.isDeprecated();
+	}
+
+	public GnuDemanglerOptions(GnuDemanglerFormat format, boolean deprecated) {
+		this.format = format;
+		this.deprecated = deprecated;
+		if (!isValidateFormat(format, deprecated)) {
+			throw new IllegalArgumentException(
+				format.name() + " is not available in the "+getDemanglerName());
+		}
 	}
 
 	public GnuDemanglerOptions(DemanglerOptions copy) {
 		super(copy);
 
 		if (copy instanceof GnuDemanglerOptions) {
 			GnuDemanglerOptions gCopy = (GnuDemanglerOptions) copy;
-			demanglerName = gCopy.demanglerName;
-			demanglerApplicationArguments = gCopy.demanglerApplicationArguments;
+			format = gCopy.format;
+			deprecated = gCopy.deprecated;
+		} else {
+			format = GnuDemanglerFormat.AUTO;
+			deprecated = false;
 		}
 	}
 
+	private GnuDemanglerOptions(GnuDemanglerOptions copy, GnuDemanglerFormat format,
+			boolean deprecated) {
+		super(copy);
+		this.format = format;
+		this.deprecated = deprecated;
+	}
+
 	/**
 	 * Returns the external demangler executable name to be used for demangling.  The 
 	 * default value is {@link #GNU_DEMANGLER_DEFAULT}.
 	 * @return the name
 	 */
 	public String getDemanglerName() {
-		return demanglerName;
+		return deprecated ? GNU_DEMANGLER_V2_24 : GNU_DEMANGLER_V2_33_1;
 	}
 
 	/**
 	 * Sets the external demangler executable name to be used for demangling
 	 * @param name the name
 	 */
+	@Deprecated(forRemoval = true)
 	public void setDemanglerName(String name) {
-		this.demanglerName = name;
+		//this.demanglerName = name;
+	}
+
+	/**
+	 * A convenience method to copy the state of this options object, changing the 
+	 * demangler executable name and demangler format to the specified values.
+	 * @param format the demangling format to use
+	 * @param deprecated true to use the deprecated gnu demangler, else false
+	 * @return the new options
+	 * @throws IllegalArgumentException if the current format is not available in the
+	 * selected demangler.
+	 */
+	public GnuDemanglerOptions withDemanglerFormat(GnuDemanglerFormat format, boolean deprecated)
+			throws IllegalArgumentException {
+		if (this.format == format && this.deprecated == deprecated) {
+			return this;
+		}
+		if (isValidateFormat(format, deprecated)) {
+			return new GnuDemanglerOptions(this, format, deprecated);
+		}
+		throw new IllegalArgumentException(
+			format.name() + " is not available in the "+getDemanglerName());
+	}
+
+	private static boolean isValidateFormat(GnuDemanglerFormat format, boolean deprecated) {
+		if (deprecated && (format.isBoth() || format.isDeprecated())) {
+			return true;
+		}
+		if (!deprecated && (format.isBoth() || format.isModernDemangler())) {
+			return true;
+		}
+		return false;
 	}
 
 	/**
 	 * Returns the current arguments to be passed to the external demangler executable
 	 * @return the arguments
 	 */
 	public String getDemanglerApplicationArguments() {
-		return demanglerApplicationArguments;
+		if (format.getFormat().isBlank()) {
```

Comment:
This is a bit confusing here.  I would make it something like `format.isAuto()` or `format.isNotSpecified()`  With the latter name you would no longer need the inline comment.

Reply:
I used `""` for auto to avoid any possible NPE and to avoid specifying "-s auto" since no format is auto anyway. It makes more sense to me to just check if it is auto.